Добавлены новые методы в InMemoryHistoryManager согласно ТЗ

Kilameen opened the pull request:
Отредактирован Класс Main согласно ТЗ
Отредактированны Методы в InMemoryTaskManage
Добавлен новый класс Node согласно ТЗ
Добавлены новые тесты

Kilameen committed:
Добавлены новые методы в InMemoryHistoryManager согласно ТЗ
Отредактирован Класс Main согласно ТЗ
Отредактированны Методы в InMemoryTaskManage
Добавлен новый класс Node согласно ТЗ
Добавлены новые тесты

Kilameen committed:
Добавлены новые методы в InMemoryHistoryManager согласно ТЗ
Отредактирован Класс Main согласно ТЗ
Отредактированны Методы в InMemoryTaskManage
Добавлен новый класс Node согласно ТЗ
Добавлены новые тесты

Kilameen committed:
Добавлены новые методы в InMemoryHistoryManager согласно ТЗ
Отредактирован Класс Main согласно ТЗ
Отредактированны Методы в InMemoryTaskManage
Добавлен новый класс Node согласно ТЗ
Добавлены новые тесты

Kilameen committed:
Добавлены новые методы в InMemoryHistoryManager согласно ТЗ
Отредактирован Класс Main согласно ТЗ
Отредактированны Методы в InMemoryTaskManage
Добавлен новый класс Node согласно ТЗ
Добавлены новые тесты

Kilameen committed:
Добавлены новые методы в InMemoryHistoryManager согласно ТЗ
Отредактирован Класс Main согласно ТЗ
Отредактированны Методы в InMemoryTaskManage
Добавлен новый класс Node согласно ТЗ
Добавлены новые тесты

Kilameen committed:
Добавлены новые методы в InMemoryHistoryManager согласно ТЗ
Отредактирован Класс Main согласно ТЗ
Отредактированны Методы в InMemoryTaskManage
Добавлен новый класс Node согласно ТЗ
Добавлены новые тесты

Kilameen committed:
Добавлены новые методы в InMemoryHistoryManager согласно ТЗ
Отредактирован Класс Main согласно ТЗ
Отредактированны Методы в InMemoryTaskManage
Добавлен новый класс Node согласно ТЗ
Добавлены новые тесты

Kilameen committed:
Добавлены новые методы в InMemoryHistoryManager согласно ТЗ
Отредактирован Класс Main согласно ТЗ
Отредактированны Методы в InMemoryTaskManage
Добавлен новый класс Node согласно ТЗ
Добавлены новые тесты

Kilameen committed:
Добавлены новые методы в InMemoryHistoryManager согласно ТЗ
Отредактирован Класс Main согласно ТЗ
Отредактированны Методы в InMemoryTaskManage
Добавлен новый класс Node согласно ТЗ
Добавлены новые тесты

Kilameen committed:
Добавлены новые методы в InMemoryHistoryManager согласно ТЗ
Отредактирован Класс Main согласно ТЗ
Отредактированны Методы в InMemoryTaskManage
Добавлен новый класс Node согласно ТЗ
Добавлены новые тесты

Kilameen committed:
Добавлены новые методы в InMemoryHistoryManager согласно ТЗ
Отредактирован Класс Main согласно ТЗ
Отредактированны Методы в InMemoryTaskManage
Добавлен новый класс Node согласно ТЗ
Добавлены новые тесты

Kilameen committed:
Добавлены новые методы в InMemoryHistoryManager согласно ТЗ
Отредактирован Класс Main согласно ТЗ
Отредактированны Методы в InMemoryTaskManage
Добавлен новый класс Node согласно ТЗ
Добавлены новые тесты

Fleaman777 commented on src/tracker/manager/InMemoryHistoryManager.java:
```diff
@@ -2,25 +2,90 @@
+
+        public Task data;
+        public Node next;
+        public Node prev;
```
Поля класса Node можно сделать private, так как доступ к ним осуществляется только внутри InMemoryHistoryManager.

Fleaman777 commented on src/tracker/manager/InMemoryTaskManager.java:
```diff
@@ -205,6 +203,7 @@ public void deleteSubtask(int id) {
             if (epic != null) {
                 epic.getSubtasksByEpic().remove((Integer) id);
                 updateStatusEpicId(epic.getId());
```
Название метода не очень понятное, лучше переименовать в updateEpicStatus.

Fleaman777 commented on test/manager/HistoryManagerTest.java:
```diff
@@ -39,9 +39,48 @@ void historyVersionTest() {
+
+        historyManager.add(task1);
+        historyManager.add(task2);
+        historyManager.add(task3);
```
хорошо бы общую часть вынести в отдельный метод, что бы не дублировать логику

Fleaman777 commented on test/manager/HistoryManagerTest.java:
```diff
@@ -39,9 +39,48 @@ void historyVersionTest() {
+        historyManager.add(task3);
+
+        historyManager.remove(task2.getId());
+        assertEquals(List.of(task1, task3), historyManager.getHistory());
```
вижу написанные новые тесты, но так же еще необходимо добавить 

- тест на создание эпика с подзадачами и пустого эпика
- проверку порядка задач в истории при повторных запросах

Kilameen committed:
Исправление замечаний по ревью.
1.Поля класса Node в InMemoryHistoryManager изменены с public на  private.
2.В классе InMemoryTaskManager, название метода  updateStatusEpicId измененно на  updateEpicStatus.
3.В классе HistoryManagerTest создан метод addTaskToHistory для упрощения кода и исключния дублирования логики.
4.В классе HistoryManagerTest добавлен новый тест checkingTheOrderOfTasksInTheHistoryForRepeatedRequests для проверки порядка задач в истории при повторных запросах.
5.В классе InMemoryTaskManagerTest добавлены два новых теста - addEpicWithTreeSubtaskTest и addEpicIsEmpty на проверку создания эпика с подзадачами и пустого эпика.

Kilameen committed:
Исправление замечаний по ревью.
1.Поля класса Node в InMemoryHistoryManager изменены с public на  private.
2.В классе InMemoryTaskManager, название метода  updateStatusEpicId измененно на  updateEpicStatus.
3.В классе HistoryManagerTest создан метод addTaskToHistory для упрощения кода и исключния дублирования логики.
4.В классе HistoryManagerTest добавлен новый тест checkingTheOrderOfTasksInTheHistoryForRepeatedRequests для проверки порядка задач в истории при повторных запросах.
5.В классе InMemoryTaskManagerTest добавлены два новых теста - addEpicWithTreeSubtaskTest и addEpicIsEmpty на проверку создания эпика с подзадачами и пустого эпика.
